src/views/components/AnalyticsGraph/index.vue

```diff
@@ -0,0 +1,312 @@
+          })
+          .on('contextmenu', function() {
+            this.dispatchEvent(new Event('drag'));
+            d3.event.preventDefault();
+          });
+
+      function brushended() {
+          let s = d3.event.selection;
+          if (!s) {
+            if (!idleTimeout) return idleTimeout = setTimeout(idled, idleDelay);
+            xScale.domain([minX, maxX]);
+            yScale.domain([minY, maxY]);
+          } else {
+            xScale.domain([s[0][0] * ratio, s[1][0]].map(xScale.invert, xScale));
+            yScale.domain([s[1][1], s[0][1] * ratio].map(yScale.invert, yScale));
+            svg.select(".brush").call(brush.move, null);
+          }
+          zoom();
+        }
+
+      function idled() {
+        idleTimeout = null;
+      }
+
+      function zoom() {
+        let t = svg.transition().duration(750);
+        svg.select(".axis--x").transition(t).call(xAxis);
+        g.select(".axis--y").transition(t).call(yAxis);
+        g.selectAll(".circles").transition(t)
+            .attr("cx", function(d) { return xScale(d.datetime); })
+            .attr("cy", function(d) { return yScale(d.amountOfProduct); });
+        g.selectAll(".line").transition(t)
+            .attr("d", function(d) { return line(d); });
+
+       voronoiDiagram = d3.voronoi()
+          .x(function(d) {return xScale(d.datetime); })
+          .y(function(d) {return yScale(d.amountOfProduct); })
+          .size([containerWidth, containerHeight])(vorData);
+      }
+
+      function dragged() {
+        d3.selectAll('.line')
+          .attr('transform', `translate(${d3.event.x}, ${d3.event.y})`);
+        d3.selectAll('.line')
+          .attr('transform', `translate(${d3.event.x}, ${d3.event.y})`);
+        g.select(".axis--x").call(xAxis);
+        g.select(".axis--y").call(yAxis);
+      }
+
+      //Graph Title from props
+      g.append("text")
+        .attr("transform",
+            "translate(" + (width/2) + " ," + 
+                           (height / 10 ) + ")")
+        .attr("text-anchor", "middle")  
+        .style("font-size", "2.2vh") 
+        .style("text-decoration", "underline")
+        .style("font-weight", "bold")
+        .text(title);
+
+      // text label for the x axis taken from props
+      g.append("text")             
+        .attr("transform",
+            "translate(" + (width/2) + " ," + 
+                           (height + (margin.top * 4)) + ")")
+        .style("text-anchor", "middle")
+        .style("font-size", "2vh") 
+        .style("font-weight", "bold")
+        .text(xUnits);
+
+      // text label for the y axis taken from props
+      g.append("text")
+          .attr("transform", "rotate(-90)")
+          .attr("y", 0 - (margin.left + 4))
+          .attr("x",0 - (height / 2))
+          .attr("dy", "1em")
+          .style("text-anchor", "middle")
+          .style("font-size", "2vh") 
+          .style("font-weight", "bold")
+          .text(yUnits); 
+    }
+  }
+};
+</script>
+<style>
```

Comment:
could you add the scoped keyword to this style tag so it doesn't effect the other styles in other components? especially since you're using styling the body tag

Reply:
I just tried adding scoped. it turns out the Vue scoped styling doesn't work with d3 properly. There are some hacky solutions in the forums but I don't think it's worth implementing. I just deleted the body styling since it wasn't needed anyway. The rest of the class names should be graph specific.